Source/WebCore/animation/WebAnimationTypes.h

```diff
@@ -64,6 +67,33 @@ using AnimationEvents = Vector<Ref<AnimationEventBase>>;
 using PropertyToTransitionMap = HashMap<CSSPropertyID, RefPtr<CSSTransition>>;
 using CSSAnimationCollection = ListHashSet<RefPtr<CSSAnimation>>;
 
+using AnimatableProperty = std::variant<CSSPropertyID, AtomString>;
+using AnimatablePropertyToTransitionMap = HashMap<AnimatableProperty, RefPtr<CSSTransition>>;
+
 } // namespace WebCore
 
+namespace WTF {
+
+template<> struct DefaultHash<WebCore::AnimatableProperty> {
+    static unsigned hash(const WebCore::AnimatableProperty& key) {
+        return WTF::switchOn(key, [] (WebCore::CSSPropertyID property) { return DefaultHash<WebCore::CSSPropertyID>::hash(property); }
+                                , [] (AtomString string) { return DefaultHash<AtomString>::hash(string); });
```

Comment:
We should try to use `const AtomString&` here to cut down on reference count churn.